app/MainHie.hs

```diff
@@ -42,29 +39,26 @@ import           Haskell.Ide.Engine.Plugin.Floskell
 
 -- | This will be read from a configuration, eventually
 plugins :: Bool -> IdePlugins
-plugins includeExamples = pluginDescToIdePlugins allPlugins
+plugins includeExamples = mkIdePlugins allPlugins
   where
     allPlugins = if includeExamples
                    then basePlugins ++ examplePlugins
                    else basePlugins
     basePlugins =
-      [ applyRefactDescriptor "applyrefact"
-      , baseDescriptor        "base"
-      , brittanyDescriptor    "brittany"
-      , buildPluginDescriptor "build"
-      , ghcmodDescriptor      "ghcmod"
-      , haddockDescriptor     "haddock"
-      , hareDescriptor        "hare"
-      , hoogleDescriptor      "hoogle"
-      , hsimportDescriptor    "hsimport"
-      , liquidDescriptor      "liquid"
-      , packageDescriptor     "package"
-      , pragmasDescriptor     "pragmas"
-      , floskellDescriptor    "floskell"
+      [ applyRefactDescriptor
+      , brittanyDescriptor
+      , ghcmodDescriptor
+      , haddockDescriptor
+      , hareDescriptor
+      , hsimportDescriptor
+      , liquidDescriptor
+      , packageDescriptor
+      , pragmasDescriptor
+      , floskellDescriptor
       ]
```

Comment:
The reason I put in the plugin id string when the plugin list is assembled is so that people can easily change the list of installed plugins, and only need to worry about possible name clashes in one place, namely here.  By delegating it to the individual plugins we lose this capability.

Unless there is some other mechanism to disambiguate them?

Reply:
I think this might need a larger discussion about how loading plugins will eventually look like.
I'm also thinking it might make more sense to remove `pluginId` as a field from `PluginDescriptor`, and have HIE generate a unique id at launch instead?

Reply:
Scrap that thought: It's used by liquid/ghc-mod and floskell/brittany to get specific plugins